edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Docker/DockerConfig.cs

```diff
@@ -166,6 +170,11 @@ public override void WriteJson(JsonWriter writer, object value, JsonSerializer s
                     writer.WriteValue(chunk);
                 }
 
+                dockerconfig.NotaryContentTrust.ForEach(ct =>
```

Comment:
This code does not use the JsonConverter attribution to achieve serialization - which would have been easier to read. It is a good opportunity to make things consistent by using [OptionConverter](https://github.com/Azure/iotedge/blob/75d74603664d1f206585ab3473294236b142a011/edge-util/src/Microsoft.Azure.Devices.Edge.Util/json/OptionConverter.cs). See [SystemModules](https://github.com/Azure/iotedge/blob/027a509549a248647ed41ca7fe1dc508771c8123/edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/SystemModules.cs) for example.

If you don't land up converting the serialization to use JsonConverters, consider using option.HasValue and always serialize and write property notaryContentTrust.